### PR TITLE
fix(dom): Ignore nulls in selectOption

### PR DESCRIPTION
**Pull request by kblok**
`selectOption` is failing if I pass nulls.

One other possibility would be assert nulls, but I like the `await page.selectOption('select', null);` to express that you want to unselect the element.

### CODE REVIEW

**Comment by pavelfeldman** [comment]
We would not want it to mask a bug where null is passed by mistake, so I would throw. But I like await `page.selectOption('select', null);` as well, so I would special-case it and treat `null` as `[]`

**Comment by kblok** [comment]
How about now @pavelfeldman ?

File: src/dom.ts [hunk 1]
```diff
@@ -332,12 +332,15 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
     this._page._log(inputLog, `elementHandle.selectOption(%s)`, values);
     const deadline = this._page._timeoutSettings.computeDeadline(options);
     let vals: string[] | ElementHandle[] | types.SelectOption[];
-    if (!Array.isArray(values))
+    if (!values)
```
Comment (dgozman) [inline src/dom.ts:335]: - Let's do a strict check `values === null`.
- Please change the method and api.md to accept null: `selectOption(values: null | string | ...`

File: src/dom.ts [hunk 2]
```diff
@@ -332,12 +332,15 @@ export class ElementHandle<T extends Node = Node> extends js.JSHandle<T> {
...
+      vals = [];
+    else if (!Array.isArray(values))
       vals = [ values ] as (string[] | ElementHandle[] | types.SelectOption[]);
     else
       vals = values;
     const selectOptions = (vals as any).map((value: any) => typeof value === 'object' ? value : { value });
     for (const option of selectOptions) {
+      assert(option, 'Value items must not be null');
```
Comment (dgozman) [inline src/dom.ts:343]: Same here: `option !== null`.

**Review by dgozman — COMMENTED** [review]
Looks good!

**Review by dgozman — APPROVED** [review]
Looks great!

**Pull request merged**